homeassistant/components/binary_sensor/nest.py

```diff
@@ -63,15 +48,22 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
         return
 
     nest = hass.data[DATA_NEST]
-    conf = config.get(CONF_MONITORED_CONDITIONS, _VALID_BINARY_SENSOR_TYPES)
 
-    for variable in conf:
+    # Add Nest Protect sensors if no binary sensor config is specified
+    if discovery_info == {}:
+        conditions = BINARY_TYPES
+    else:
+        conditions = discovery_info.get(CONF_MONITORED_CONDITIONS, {})
+        # Add all binary sensors if no monitored conditions are specified
+        if all(c is None for c in conditions):
```

Comment:
This is needed to allow users adding all available sensors by setting a blank monitored conditions list. The functionality already existed but wasn't working properly. If that is done the get for monitored conditions returns a list with none {none}. 

Maybe it is better to allow this functionality by not setting a monitored condition list at all?